src/index.js

```diff
@@ -0,0 +1,182 @@
+          <Dropdown
+			position="bottom left"
+			contentClassName="edit-post-hm-post-repeat__dialog"
+			renderToggle={ ( { isOpen, onToggle } ) => (
+				<Button
+					aria-expanded={ isOpen }
+					className="edit-post-hm-post-repeat__toggle"
+					onClick={ onToggle }
+					isLink
+				>
+					<HMPostRepeatFields />
+					
+				</Button>
+			) }
+			renderContent={ () => (
+				<fieldset
+					key="hm-post-repeat-selector"
+					className="hm-post-repeat__dialog-fieldset"
+				>
+					<legend className="hm-post-repeat__dialog-legend">
+						{ __( 'Repeat Frequency' ) }
+					</legend>
+					
+					<div className="hm-post-repeat__choice">
+						<input
+							type="radio"
+							name={ `hm-post-repeat__setting-no` }
+							value=""
+							className="hm-post-repeat__dialog-radio"
+						/>
+						<label
+							htmlFor={ `hm-post-repeat-no-` }
+							className="hm-post-repeat__dialog-label"
+						>
+							No
+						</label>
+						<p>
+								No repeat selected
+						</p>
+					</div>
+					<div className="hm-post-repeat__choice">
+						<input
+							type="radio"
+							name={ `hm-post-repeat__setting-` }
+							value=""
+							className="hm-post-repeat__dialog-radio"
+						/>
+						<label
+							htmlFor={ `hm-post-repeat-no-` }
+							className="hm-post-repeat__dialog-label"
+						>
+							Daily
+						</label>
+						<p>
+								Information description
+						</p>
+					</div>
+					<div className="hm-post-repeat__choice">
+						<input
+							type="radio"
+							name={ `hm-post-repeat__setting-` }
+							value=""
+							className="hm-post-repeat__dialog-radio"
+						/>
+						<label
+							htmlFor={ `hm-post-repeat-no-` }
+							className="hm-post-repeat__dialog-label"
+						>
+							Weekly
+						</label>
+						<p>
+								Information description
+						</p>
+					</div>
+					<div className="hm-post-repeat__choice">
+						<input
+							type="radio"
+							name={ `hm-post-repeat__setting-` }
+							value=""
+							className="hm-post-repeat__dialog-radio"
+						/>
+						<label
+							htmlFor={ `hm-post-repeat-no-` }
+							className="hm-post-repeat__dialog-label"
+						>
+							Fortnightly
+						</label>
+						<p>
+								Every 2 weeks
+						</p>
+					</div>
+					<div className="hm-post-repeat__choice">
+						<input
+							type="radio"
+							name={ `hm-post-repeat__setting-` }
+							value=""
+							className="hm-post-repeat__dialog-radio"
+						/>
+						<label
+							htmlFor={ `hm-post-repeat-no-` }
+							className="hm-post-repeat__dialog-label"
+						>
+							Monthly
+						</label>
+						<p>
+								Information description
+						</p>
+					</div>
+					
+				</fieldset>
+			) }
+		/>
+        </PluginPostStatusInfo>
+     </>
+    )
+  }
+})
+
+let HMPostRepeatFields = (props) => {
+	return (
+		<>
+			{props.hmpostrepeat_metafield}
+		</>
+	)
+}
+
+// Grab stored meta value of hm-post-repeat using a redux store?!
+HMPostRepeatFields: withSelect(
```

Comment:
This isn't really valid JS, or it at least won't be doing anything, you do this kind of assignment within an object like:
```js
const thing = {
  foo: 'bar'
};
```

I think for what you're trying to do there's a common pattern where you create a component and then you enhance that component. They don't need to have the same variable name. For example:

```jsx
const MyComponent = props => {
  return <p>repeat every {props.repeat}</p>
};

const MyEnhancedComponent = withSelect( select => {
  return {
    repeat: select('core/editor').getEditedPostAttribute('meta')['hm-post-repeat']
  }
} )( MyComponent );
```

Reply:
the `withSelect` can be refactored away with `useSelect`

```js
const HMPostRepeatFields = () => {
	const { repeat } = useSelect( ( select ) => {
		return { repeat: select('core/editor').getEditedPostAttribute('meta')['hm-post-repeat']
	}, [ ] );
	if ( 'fortnightly' === repeat ) {
...
```

Reply:
something like this:

```js

const PostRepeating = () => {
	const { repeat } = useSelect(
		( select ) => { repeat: select('core/editor').getEditedPostAttribute('meta')['hm-post-repeat'] },
		[ ]
	);
	const { editPost } = useDispatch( 'core/editor' );
	const setRepeating = useCallback( ( newRepeat ) => {
		editPost( {
			meta: {
				'hm-post-repeat': newRepeat
			}
		} );
	}, [ repeat ] );

	const labels = [
		'fortnightly' => 'Fortnightly',
		'weekly'      => 'Weekly'
	];
	const values = Object.keys( labels );
	return values.map( value => {
		<label key={`hm-repeat${value}`}>
			<input type="radio" value={value} name="hm-repeat"  onClick={ setRepeating( value ) } />
			{ labels[value] }
		</label>
	});
}
```

Reply:
With that you can eliminate most of the other JS

Reply:
can probably be simpllified further by using a `SelectControl` https://developer.wordpress.org/block-editor/reference-guides/components/select-control/

Reply:
and GB's version of radio buttons https://developer.wordpress.org/block-editor/reference-guides/components/radio-group/